moveit_core/distance_field/include/moveit/distance_field/propagation_distance_field.h

```diff
@@ -441,9 +442,9 @@ class PropagationDistanceField : public DistanceField
     return max_distance_sq_;
   }
 
+  typedef std::vector<Eigen::Vector3i, Eigen::aligned_allocator<Eigen::Vector3i>> StlVectorEigenVector3i;
 private:
-  typedef std::set<Eigen::Vector3i, compareEigen_Vector3i> VoxelSet; /**< \brief Typedef for set of integer indices */
-
+  typedef std::set<Eigen::Vector3i, compareEigen_Vector3i, Eigen::aligned_allocator<Eigen::Vector3i>> VoxelSet; /**< \brief Typedef for set of integer indices */
```

Comment:
The integer datastructures and sets are not defined in `eigen_stl_containers` (yet).
I guess it would make sense to contribute them there, although we might still provide them on our own for now.

Incidentally, we might want to consider using `EIGEN_DEFINE_STL_VECTOR_SPECIALIZATION` et. al. from `Eigen/StdVector` in the future.

Reply:
What do you prefer for now?
`EIGEN_DEFINE_STL_VECTOR_SPECIALIZATION(Eigen::Vector3i)` or the ``typedef std::vector<Eigen::Vector3i, Eigen::aligned_allocator<Eigen::Vector3i>> StlVectorEigenVector3i;`` ?

Personally, I prefer `typedef` since it makes it more easy to check if the container is aligned in bigger projects ...

Reply:
We don't use the eigen macro anywhere else yet. It's probably better to stay consistent with typedefs.

The macros, together with typedef support for other std types (sets) would make a great addition to the eigen_std_containers package though.